enterprise/frontend/src/embedding-sdk/store/collections.ts

```diff
@@ -16,13 +13,13 @@ export type SDKCollectionReference = RegularCollectionId | "personal" | "root";
 export const getCollectionNumericIdFromReference = createSelector(
   [
     getUserPersonalCollectionId,
-    (_, collectionReference: SDKCollectionReference) => collectionReference,
+    (_, collectionReference: SdkCollectionId) => collectionReference,
   ],
-  (personalCollectionId, collectionReference) => {
+  (personalCollectionId, collectionReference): CollectionId | null => {
     return match(collectionReference)
       .with("personal", () => personalCollectionId as RegularCollectionId)
       .with("root", () => null)
-      .with(P.number, () => collectionReference)
+      .with(P.number, () => collectionReference as CollectionId)
```

Comment:
I don't think this make sense. We're matching a number, but we return the type as `CollectionId` which could also be a string
https://github.com/metabase/metabase/blob/22ecdaa9bb3e1f5aa979bcbbc7aeda581fd9d8c2/frontend/src/metabase-types/api/collection.ts#L21-L26

I think we should return `RegularCollectionId`

```suggestion
      .with(P.number, () => collectionReference as RegularCollectionId)
```

This happens on line 46 as well.